cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -0,0 +1,132 @@
+#include <iostream>
+#include <memory>
+#include <list>
+#include <algorithm>
+
+class TimedEntry {
+public:
+    size_t count() const { return m_count; }
+    double time() const { return m_time; }
+
+    TimedEntry& add_time(double time) {
+        ++m_count;
+        m_time += time;
+        return *this;
+    }
+
+private:
+    size_t m_count = 0;
+    double m_time = 0.0;
+};
+
+class RadixTreeNode {
+private:
+    using ChildList = std::list<std::unique_ptr<RadixTreeNode>>;
+
+public:
+    RadixTreeNode(std::string name) : key(std::move(name)),  prev(nullptr) {}
+    
+    const std::string& getKey() const { return key; }
+    TimedEntry& getData() { return data; }
+    const TimedEntry& getData() const { return data; }
+    const ChildList& getChildren() const { return children; }
+
+    // Add a child node
+    RadixTreeNode* addChild(std::string childKey) {
+        auto newChild = std::make_unique<RadixTreeNode>(std::move(childKey));
+        newChild->prev = this;
+        children.push_back(std::move(newChild));
+        return children.back().get();
+    }
+
+    // Get previous node (parent) in the tree
+    RadixTreeNode* getPrev() const {
+        return prev;
+    }
+
+private:
+    std::string key;
```

Comment:
Is this key used for indexing?

String comparison is slow and may not be good for indexing.

Reply:
How about mapping a function name (std::string) to a unsigned int.
And use the unsigned int as key to do the comparison?

Reply:
Done.